File: MojPrijevozFront/moj_prijevoz/lib/resources/search_objects/fare/fare_search_object.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:moj_prijevoz/common/resources/profile_type.dart';
import 'package:moj_prijevoz/common/resources/search_objects/base_search_object.dart';

part 'fare_search_object.g.dart';

@JsonSerializable()
class FareSearchObject extends BaseSearchObject {
  ProfileType fareRole;
  int? fareId;
  FareSearchObject({
    required this.fareRole,
    required super.page,
    required super.pageSize,
    this.fareId,
  });

  @override
  Map<String, dynamic> toJson() => _$FareSearchObjectToJson(this);

  factory FareSearchObject.fromJson(Map<String, dynamic> json) =>
      _$FareSearchObjectFromJson(json);
}

File: MojPrijevoz/MojPrijevoz.Model/SearchObjects/FareSearchObject.cs
using MojPrijevoz.Database;
using MojPrijevoz.Model.BaseModels;

namespace MojPrijevoz.Model.SearchObjects;

public class FareSearchObject : BaseSearchObject  {
    public int? FareId { get; set; }
    public ProfileType FareRole { get; set; }
}